domains/zeto/internal/zeto/handler_events.go

```diff
@@ -149,6 +151,36 @@ func (z *Zeto) handleLockedEvent(ctx context.Context, smtTree *merkleTreeSpec, s
 	return nil
 }
 
+func (z *Zeto) handleIdentityRegisteredEvent(ctx context.Context, smtKycTree *common.MerkleTreeSpec, ev *prototk.OnChainEvent, tokenName string, res *prototk.HandleEventBatchResponse) error {
+	var registered IdentityRegisteredEvent
+	if err := json.Unmarshal([]byte(ev.DataJson), &registered); err == nil {
+		txData, err := decodeTransactionData(ctx, registered.Data)
+		if err != nil || txData == nil {
+			txId := pldtypes.MustParseBytes32("0000000000000000000000000000000000000000000000000000000000000000")
+			log.L(ctx).Infof("IdentityRegistered event [publicKey=%+v] has no tx data. Inserting zero tx ID", &registered.PublicKey)
+			txData = &types.ZetoTransactionData_V0{
+				TransactionID: txId,
+				InfoStates:    []pldtypes.Bytes32{},
+			}
+		}
+		z.recordTransactionInfo(ev, txData, res)
```

Comment:
I'm not sure there's any transaction info to record, especially if txID might be 0? Do we expect "decodeTransactionData" to actually decode something or not (or we're unsure depending on the transaction)?

Reply:
indeed. the call is removed.